dm/worker/subtask.go

```diff
@@ -163,34 +164,50 @@ func (st *SubTask) Run() {
 
 // fetchResult fetches units process result
 // when dm-unit report an error, we need to re-Process the sub task
-func (st *SubTask) fetchResult(ctx context.Context, cancel context.CancelFunc, pr chan pb.ProcessResult) {
+func (st *SubTask) fetchResult(pr chan pb.ProcessResult) {
 	defer st.wg.Done()
 
+retry:
 	select {
-	case <-ctx.Done():
+	case <-st.ctx.Done():
 		return
 	case result := <-pr:
 		st.setResult(&result) // save result
-		cancel()              // dm-unit finished, canceled or error occurred, always cancel processing
+		st.cancel()           // dm-unit finished, canceled or error occurred, always cancel processing
 
 		if len(result.Errors) == 0 && st.Stage() == pb.Stage_Paused {
 			return // paused by external request
 		}
 
-		var stage pb.Stage
+		var (
+			cu    = st.CurrUnit()
+			stage pb.Stage
+		)
 		if len(result.Errors) == 0 {
 			if result.IsCanceled {
 				stage = pb.Stage_Stopped // canceled by user
 			} else {
 				stage = pb.Stage_Finished // process finished with no error
 			}
 		} else {
+			/* TODO
+			it's a poor and very rough retry feature, the main reason is that
+			the concurrency control of the sub task module is very confusing and needs to be optimized.
```

Comment:
Is the `concurrency control optimization` including the following scenario?
all of DM-worker's connections to downstream TiDB suddenly reset by downstream(often caused by network issue), then we must recover each connection in syncer and if the worker-count config of syncer is N, we will wait (N + 1) * 10 seconds at most.

Reply:
first of all it is concurrent, I don't understand why we need to wait (N + 1) * 10 seconds at most.
For the refactoring, more is the reconstruction of the the structure, making the moudle and funcion more reasone, of course we can increase the backoff property

Reply:
I mean for each subtask, `(N + 1) * 10 seconds` may be not continuous, but each DB connection must be recovered. So the worst case is `(N + 1) * 10 seconds` sums up.

Reply:
but it's real concurrent, right?

Reply:
it's the logic of unit, we can refine while refactoring  sync unit

Reply:
we can refactor units one by one and then the outer state transition later.